Amazon.Ion.ObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -114,6 +115,22 @@ public void SerializesAndDeserializesSubtypesBasedOnTypeAnnotations()
             );
         }
 
+        [TestMethod]
+        public void SerializeMaxDepthException()
+        {
+            var serializer = new IonSerializer(new IonSerializationOptions {MaxDepth = 3});
+            Assert.ThrowsException<NotSupportedException>(() => serializer.Serialize(TestObjects.honda));
```

Comment:
MaxDepth only applies to Deserialization

Reply:
Removed MaxDepth logic from serialization.